utils/utils.go

```diff
@@ -261,7 +263,7 @@ func unmarshalJSONFiles(path string, ignoreDirErr bool) ([]interface{}, error) {
 func unmarshalJSONFilesWithFilenames(path string) (map[string]interface{}, error) {
 	fileInfos, err := ioutil.ReadDir(path)
 	if err != nil {
-		return nil, err
+		return nil, errEnvironmentsDirNotFound
```

Comment:
i think its fair to say we want to avoid any "environments-specific" logic to be held inside a helper function like `unmarshalJSONFilesWithFilenames` here

we can keep the `return nil, err` here and push this type of check up to where you call it around L#230.  i _think_ that the `ioutil.ReadDir` method should return an error that could be checked with `os.IsNotExist` (a quick test should prove that out), otherwise you can always just call `_, err := os.Stat(environmentsDir)` and inspect that one instead to verify the directory's existence or not

Reply:
Oh yeah good point, you're right that I shouldn't have put that into a util function.

Sadly, doing `os.IsNotExists` doesn't work, so I'll just first check for existence on environments